vertical-pod-autoscaler/pkg/updater/priority/update_priority_calculator_test.go

```diff
@@ -336,37 +336,131 @@ func TestDontUpdatePodWithOOMAfterLongRun(t *testing.T) {
+		{
+			name:       "container listed in VpaOvservedContainers annotation",
+			annotation: map[string]string{annotations.VpaObservedContainersLabel: containerName},
+			want:       true,
+		},
+		{
+			// Continers not liseted in VpaOvservedContainers annotation
+			// shouldn't trigger the quick OOM.
+			name:       "container not listed in VpaOvservedContainers annotation",
+			annotation: map[string]string{annotations.VpaObservedContainersLabel: ""},
+			want:       false,
 		},
-		{},
 	}
+	for _, tc := range tests {
+		t.Run(fmt.Sprintf("test case: %s", tc.name), func(t *testing.T) {
+			calculator := NewUpdatePriorityCalculator(
+				nil, nil, &UpdateConfig{MinChangePriority: 0.5}, &test.FakeRecommendationProcessor{})
+
+			pod := test.Pod().WithAnnotations(tc.annotation).
+				WithName("POD1").AddContainer(test.BuildTestContainer(containerName, "4", "")).Get()
+
+			// Pretend that the test pod started 11 hours ago.
+			timestampNow := pod.Status.StartTime.Time.Add(time.Hour * 11)
+
+			pod.Status.ContainerStatuses = []apiv1.ContainerStatus{
+				{
+					Name: containerName,
+					LastTerminationState: apiv1.ContainerState{
+						Terminated: &apiv1.ContainerStateTerminated{
+							Reason:     "OOMKilled",
+							FinishedAt: metav1.NewTime(timestampNow.Add(-1 * 3 * time.Minute)),
+							StartedAt:  metav1.NewTime(timestampNow.Add(-1 * 5 * time.Minute)),
+						},
+					},
+				},
+			}
+
+			// Pod is within the recommended range.
+			recommendation := test.Recommendation().WithContainer(containerName).
+				WithTarget("5", "").
+				WithLowerBound("1", "").
+				WithUpperBound("6", "").Get()
+
+			calculator.AddPod(pod, recommendation, timestampNow)
+			result := calculator.GetSortedPods(NewDefaultPodEvictionAdmission())
+			isUpdate := len(result) != 0
+			assert.Equal(t, tc.want, isUpdate)
+		})
+	}
+}
 
-	// Pod is within the recommended range.
-	recommendation := test.Recommendation().WithContainer(containerName).
-		WithTarget("5", "").
-		WithLowerBound("1", "").
-		WithUpperBound("6", "").Get()
-
-	calculator.AddPod(pod, recommendation, timestampNow)
-	result := calculator.GetSortedPods(NewDefaultPodEvictionAdmission())
-	assert.Exactly(t, []*apiv1.Pod{}, result, "Pod shouldn't be updated")
+func TestDontUpdatePodWithOOM_ContainerScalingMode(t *testing.T) {
+	tests := []struct {
+		name        string
+		scalingMode vpa_types.ContainerScalingMode
+		want        bool
+	}{
+		{
+			name:        "ContainerScalingModeAuto",
+			scalingMode: vpa_types.ContainerScalingModeAuto,
+			want:        true,
+		},
+		{
+			// Containers with ContainerScalingModeOff
+			// shouldn't trigger the quick OOM.
+			name:        "ContainerScalingModeOff",
+			scalingMode: vpa_types.ContainerScalingModeOff,
+			want:        false,
+		},
+	}
+	for _, tc := range tests {
+		t.Run(fmt.Sprintf("test case: %s", tc.name), func(t *testing.T) {
+			resourcePolicy := &vpa_types.PodResourcePolicy{
+				ContainerPolicies: []vpa_types.ContainerResourcePolicy{
```

Comment:
You might want to add a test for default policies:
ContainerResourcePolicy = {
						ContainerName: "*",
						Mode:          &tc.scalingMode,
					},

Reply:
Tests added for `ContainerScalingModeAuto` and `ContainerScalingModeOff` as default.